File: src/main/kotlin/exam8/Exam8.java
package exam8;

public class Exam8 {

	public static void main(String[] args) {
		String name = null;

		if (name != null) {
			System.out.println(name.length());
		} else {
			System.out.println("Name is null");
		}

		// 자바에서 안전하게 함수 호출하려면 이렇게 해야 함
		// Person person = new Person();
		// if ( person != null && person.leftArm != null && person.leftArm.hand != null ) person.leftArm.hand.grab();
	}
}

File: src/main/kotlin/exam8/Exam8.kt
package exam8

class Person(
    val leftArm: Arm = Arm(),
    val rightArm: Arm = Arm()
)

class Arm(
    val hand: Hand = Hand()
)

class Hand() {
    fun grab() {
        println("Grabbing!")
    }
}

fun main() {
    val name: String? = "hello"
    println(name?.length ?: "Name is null")

    val person = Person()
    // Java 에서 안전하게 함수를 호출하려면 이렇게 해야 한다
    // if ( person != null && person.leftArm != null && person.leftArm.hand != null ) person.leftArm.hand.grab();
    person?.leftArm?.hand?.grab()
}